metadata/src/main/java/org/apache/kafka/controller/ReplicationControlManager.java

```diff
@@ -349,6 +357,16 @@ public void replay(PartitionChangeRecord record) {
         log.debug("Applied ISR change record: {}", record.toString());
     }
 
+    public void replay(RemoveTopicRecord record) {
+        TopicControlInfo topic = topics.remove(record.topicId());
+        if (topic == null) {
+            throw new RuntimeException("Can't find topic with ID " + record.topicId() +
+                " to remove.");
+        }
+        topicsByName.remove(topic.name);
```

Comment:
Should we update brokersToIsrs too?

Reply:
The ISRs should already have been updated by `BrokerChangeRecords` that were previously replayed.

Reply:
Hmm, you mean PartitionChangeRecord? I don't see PartitionChangeRecord being generated from the topicDeletion request.

Reply:
Sorry, you're right: we need to remove this from `brokersToIsrs`.  Fixed.